pysat/_instrument.py

```diff
@@ -2072,94 +2402,211 @@ def __iter__(self):
         """
 
         if self._iter_type == 'file':
+            width = self._iter_width
             for fname in self._iter_list:
-                self.load(fname=fname)
-                yield self
+                # Without a copy, a = [inst for inst in inst] leads to
+                # every item being the last day loaded.
+                # With the copy, behavior is as expected. Making a copy
+                # of an empty object is going to be faster than a full one.
+                self.data = self._null_data
+                friend = self.copy()
```

Comment:
What is `friend` supposed to be?  Is it a local copy of Instrument?  Perhaps a different variable name like `local_inst` or something that is more appropriate if I misinterpreted the code would be clearer.